### ISSUE TITLE
False positive `invalid-type-form` on union of classes with paramspec

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ty#3332]
### Summary

Found while messing around
https://play.ty.dev/47469640-f492-4c7d-bc35-9286902bc045
```py
from typing import reveal_type

if input():
    class A[**P]: ...
else:
    class A[**P]: ...

reveal_type(A)  # Revealed type: `<class 'A'> | <class 'A'>`

# Error: Bare ParamSpec `P` is not valid in this context in a parameter annotation (invalid-type-form)
def foo[**P](x: A[P]): ...
```
It works fine with a normal generic instead of a paramspec: https://play.ty.dev/5be13359-4308-46f9-a49d-75bd9714b9e8

### Version

ty 0.0.32 playground 81a81d21e
